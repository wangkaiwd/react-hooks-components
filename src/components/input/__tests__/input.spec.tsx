import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Input from "../input";

describe("Input", () => {
  it("should render correctly", () => {
    const input = renderer.create(<Input placeholder="Please input something..." />).toJSON();
    expect(input).toMatchSnapshot();
  });
  it("should clear text within input", () => {
    const onChange = jest.fn();
    render(<Input onChange={onChange} value="20" />);
    fireEvent.click(screen.getByTestId("clear"));
    expect(onChange).toBeCalledTimes(1);
    expect(screen.getByTestId("input")).toHaveValue("");
  });
  it("should trigger change with parameter value", () => {
    const onChange = jest.fn();
    render(<Input onChange={onChange} />);
    const input = screen.getByTestId("input");
    // only need to trigger input event
    fireEvent.change(input, { target: { value: "10" } });
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(input).toHaveValue("10");
  });
  it("should support all attributes of native input", () => {
    const placeholder = "Please input";
    const maxLength = 10;
    const { container } = render(<Input placeholder={placeholder} maxLength={maxLength} />);
    const input = container.querySelector("input");
    expect(input?.getAttribute("placeholder")).toBe(placeholder);
    expect(input?.getAttribute("maxlength")).toBe(String(maxLength));
  });
});
