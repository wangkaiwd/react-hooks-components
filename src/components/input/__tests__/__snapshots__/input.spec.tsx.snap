// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div
  className="ant-input"
>
  <input
    data-testid="input"
    onChange={[Function]}
    placeholder="Please input something..."
  />
</div>
`;
